examples/oneapi/cpp/source/connected_components/connected_components_batch.cpp

```diff
@@ -0,0 +1,56 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <memory>
+
+#include "example_util/utils.hpp"
+#include "oneapi/dal/algo/connected_components.hpp"
+#include "oneapi/dal/graph/undirected_adjacency_vector_graph.hpp"
+#include "oneapi/dal/io/graph_csv_data_source.hpp"
+#include "oneapi/dal/io/load_graph.hpp"
+
+namespace dal = oneapi::dal;
+using namespace dal::preview::connected_components;
+
+int main(int argc, char** argv) {
+    const auto filename = get_data_path("graph.csv");
+
+    // read the graph
+    const dal::preview::graph_csv_data_source ds(filename);
+    const dal::preview::load_graph::descriptor<> d;
+    const auto my_graph = dal::preview::load_graph::load(d, ds);
+    std::allocator<char> alloc;
+
+    // set algorithm parameters
+    const auto cc_desc =
+        descriptor<float, method::afforest, task::vertex_partitioning, std::allocator<char>>(alloc);
```

Comment:
Use template deduction for shorter call like
`const auto cc_desc = descriptor<>(alloc);`
And what if not using connected components namespace by writing longer name?
`const auto cc_desc = **connected_components::**descriptor<>(alloc);`